src/engraving/dom/fret.cpp

```diff
@@ -84,6 +105,26 @@ static const String blankPattern(int strings)
     return pattern;
 }
 
+static HarmonyMapKey createHarmonyMapKey(const String& harmony, const ParsedChord& chord, const MStyle& style)
+{
+    NoteCaseType noteCase;
+    size_t idx;
+    NoteSpellingType spellingType = style.styleV(Sid::chordSymbolSpelling).value<NoteSpellingType>();
```

Comment:
How will the harmony key search behave if the user changes the style?

(Optimization: it’s better to move the retrieval of spellingType outside the loop over harmonyToDiagramMap, similar to what was done for chordList.)

Reply:
Good point - I've updated `NotationInteraction::addFretboardDiagram` to use `Harmony::harmonyName` which takes into account the spelling set in styles.